### PR TITLE
ControllerInterface: Exposse DSU client battery level as an input.

### PR DESCRIPTION
**Pull request by jordan-woyak**
Tested with an Android DSU Server.

I've named the input "Battery". It should be visible in the "Advanced" mapping dialog.
If someone prefers "Battery Level" or something else I don't have a strong opinion.

This is mainly intended to be used with PR #8417 to pass-through the battery level to an emulated Wii remote.

### CODE REVIEW

File: Source/Core/InputCommon/ControllerInterface/Device.h [hunk 1]
```diff
@@ -22,11 +22,18 @@ namespace ciface
...
 // TODO: Make this configurable.
 constexpr int RUMBLE_PERIOD_MS = 10;
+
 // This needs to be at least as long as the longest rumble that might ever be played.
 // Too short and it's going to stop in the middle of a long effect.
 // Infinite values are invalid for ramp effects and probably not sensible.
 constexpr int RUMBLE_LENGTH_MS = 1000 * 10;
 
+// All inputs (other than accel/gyro) return 1.0 as their maximum value.
+// Battery inputs will almost always be mapeed to the "Battery" setting which is a percentage.
```
Comment (BhaaLseN) [inline Source/Core/InputCommon/ControllerInterface/Device.h]:
```suggestion
// Battery inputs will almost always be mapped to the "Battery" setting which is a percentage.
```

Comment (jordan-woyak) [reply Source/Core/InputCommon/ControllerInterface/Device.h]: Oops. :P

**Comment by AdmiralCurtiss** [comment]
Would it make more sense to define 1.0 as maximum battery on the emulated Wii remote side so we don't have to special case the input here?

**Comment by jordan-woyak** [comment]
> Would it make more sense to define 1.0 as maximum battery on the emulated Wii remote side so we don't have to special case the input here?

I've mentioned in the comment above the `constexpr` defintion why I think 100 is a better decision.
No one is ever going to normally map this "Input" to a button press and if they do that can be smart enough to divide by 100.
People are going to map this to the "Battery" setting 99% of the time which will expect 100 as the maximum value. Multiplying by 100 will be annoying and confusing to users.

**Comment by AdmiralCurtiss** [comment]
No, what I mean is, why have the 'Battery' setting expect 100 as the maximum value instead of 1.

**Comment by jordan-woyak** [comment]
> No, what I mean is, why have the 'Battery' setting expect 100 as the maximum value instead of 1.

Oh.. :P

Well.. the setting already exists as a percentage. I think that's a more friendly interface for people not doing battery "pass through". It would require breaking configs or even more special-case handling to change it now. I think this is the lesser evil. I don't think it's too horrible to have a very specific kind of input take on a different maximum value. Accelerometer and gyroscope inputs return m/s/s and rad/s, similarly.

**Comment by AdmiralCurtiss** [comment]
Ah, fair enough, I didn't realize that Battery setting already existed. In that case yeah this is probably fine.

**Comment by rlnilsen** [comment]
@jordan-woyak I'm glad _you_ did this, as my brain is mush right now (and I don't know how long it will last). I've looked over the changes and there's nothing jumping out at me, although that might not be anything to go by.

**Comment by mbc07** [comment]
Tested with both DS4 and Wiimote DSU Servers and works as expected. Sweet!

**Comment by JMC47** [comment]
This needs to be rebased now.

**Pull request merged**